userapi/src/main/java/fk/prof/userapi/util/StackLineParser.java

```diff
@@ -0,0 +1,122 @@
+package fk.prof.userapi.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** A utility class which parses and converts Java names from their corresponding JVM Types signatures
+ - from string of format className#methodName (argTypes)retType
+ * Created by rohit.patiyal on 26/04/17.
+ */
+public class StackLineParser {
+    private static Map<Character, String> signToPrimitiveMap = new HashMap<Character, String>(){
+        {
+            put('V',"void");
+            put('Z',"boolean");
+            put('B',"byte");
+            put('C',"char");
+            put('S',"short");
+            put('I',"int");
+            put('J',"long");
+            put('F',"float");
+            put('D',"double");
+        }
+    };
+
+    private static List<JVMTypeSignatureParser> parsers = new ArrayList<JVMTypeSignatureParser>(){
+        {
+            add(StackLineParser::parseTerminalType);
+            add(StackLineParser::parseMethodName);
+            add(StackLineParser::parseArgs);
+            add(StackLineParser::parseRet);
+        }
+    } ;
+
+    public static String convertJVMTypeSignToJava(String text){
+        StringBuilder sb = new StringBuilder();
+        int textIndex = 0;
+        int sbLength = 0;
+        text = text.replace('/','.');
+        for(JVMTypeSignatureParser parser : parsers){
```

Comment:
The idea of having a list is to separate and reuse the logic to parse the different part of the stackLine instead of having a large function with a lot of if else switches. The partitions are as follows:
 1. ClassName (A Type)
 2. MethodName
 3. Arguments (Collection of Types of the arguments enclosed in '(' and ')'
 4. Return Type (A Type)
Also, given that our input string is in a specific format, we need to enforce the order of the parsing methods somewhere or our single parser would capture a wider language, if just start markers are used to determine the parsing logic. 

With this particular way we can also preserve partial successful results if in case a failure is obtained while parsing the later part of the string.